extensions-core/druid-pac4j/src/main/java/org/apache/druid/security/pac4j/Pac4jAuthenticator.java

```diff
@@ -141,7 +143,19 @@ private Config createPac4jConfig(OIDCConfig oidcConfig)
 
     OidcClient oidcClient = new OidcClient(oidcConf);
     oidcClient.setUrlResolver(new DefaultUrlResolver(true));
-    oidcClient.setCallbackUrlResolver(new NoParameterCallbackUrlResolver());
+    if (pac4jCommonConfig.getRedirectParam() != null){
+      oidcClient.setCallbackUrlResolver(
+        new NoParameterCallbackUrlResolver() {
```

Comment:
I'm not sure there's any benefit in overriding the internal implementation of this class. It would be cleaner to implement the `CallbackUrlResolver` interface directly.

Reply:
Yes agree. will take care of it.

Reply:
Done